src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c

```diff
@@ -2397,10 +2398,13 @@ int wm_vuldet_check_agent_vulnerabilities(wm_vuldet_t *vuldet) {
             // Collect agent software
             if (OS_SUCCESS != wm_vuldet_collect_agent_software(agents_it, db, &scan_ctx)) {
                 agents_it->pending_attempts--;
-                if (agents_it->pending_attempts)
+                if (agents_it->pending_attempts) {
                     retry_agents = true;
-                else
-                    mtwarn(WM_VULNDETECTOR_LOGTAG, VU_GET_SOFTWARE_ERROR, scan_ctx.agent_id, WM_VULNDETECTOR_MAX_AGENT_SCAN_ATTEMPS);
+                    if (!last_fail_scan) last_fail_scan = time(NULL);
```

Comment:
Shouldn't we take the time of the first failed scan? The thing is, if we have a big number of agents, let's say 1000, and Vulnerability Detector has the next results:

- It fails to scan the first ten agents.
- Then it successfully scans the next ninety agents.

This will give time enough to the first agent to complete the **syscollector** synchronization. The same in the opposite:

- It successfully scans the first ten agents.
- Then it fails to scan the next ninety agents.

The ninety failures will consume some time, so when you compute the time to sleep, if you make use of the first failure time, it wil probably reduce the time to sleep in (let's say...) ten seconds.

What do you think?

Reply:
This mechanism is considered in line 2433 when the time to sleep is calculated using **last_fail_scan** and **vuldet->retry_interval**.
But, the name **last_fail_scan** doesn´t help to understand this and probably should be renamed to **first_fail_scan**.
Will do!

Reply:
Great!